gcp/modules/locust/client.rb

```diff
@@ -2,79 +2,90 @@
 require "json"
 require "google/cloud/monitoring"
 
-@project_id = ENV['PROJECT_ID']
+module LocustClient
 
-def process_locust_result(locust_stats_file, locust_distribution_file, app_name)
+  @project_id = ENV['PROJECT_ID']
 
-  metrics = {}
-  all_stats = JSON.parse(File.read(locust_stats_file))
+  def self.collect_metrics(all_stats, all_distributions, user_count)
+    metrics = {}
 
-  metrics["user_count"] = all_stats["user_count"]
+    metrics["user_count"] = user_count
 
-  stats = all_stats["stats"].select do |stat|
-    stat["name"] == "Total"
-  end
+    stats = all_stats["stats"].select do |stat|
+      stat["name"] == "Total"
+    end
 
-  stats = stats.first
-  ["median_response_time", "min_response_time", "max_response_time",
-   "avg_response_time", "num_failures", "num_requests", "current_rps"].each do |metric|
-    metrics[metric] = stats[metric]
-  end
+    stats = stats.first
+    ["median_response_time", "min_response_time", "max_response_time",
+     "avg_response_time", "num_failures", "num_requests", "current_rps"].each do |metric|
+      metrics[metric] = stats[metric]
+    end
 
-  distributions = CSV.parse(File.read(locust_distribution_file))
-  distributions = distributions.select do |dist|
-    dist.include? "None Total"
-  end
+    distributions = all_distributions.select do |dist|
+      dist.include? "Total"
+    end
 
-  # Only add distributions if we actually have any
-  if distributions
-    distributions = distributions.reverse.first
-    ["100th_percentile", "99th_percentile", "98th_percentile", "95th_percentile", "90th_percentile",
-     "80th_percentile", "75th_percentile", "66th_percentile", "50th_percentile"].each do |metric|
-      metrics[metric] = distributions.pop if metrics.key?(metric)
+    # Only add distributions if we actually have any
+    unless distributions.empty?
```

Comment:
Is this like idiomatic Ruby way? Or is there some desired difference in behavior? To me it seems like "double negation" (`unless` -> `if not`, `distributions.empty?` -> `not distributions`, so the whole is like `if not not distributions`).

*note: just being curious here, rather than strongly prefering one way or another*

Reply:
For `if` vs `unless`: yes, `unless` is idiomatic: https://github.com/rubocop-hq/ruby-style-guide#unless-for-negatives

If you are proposing to use `distributions` (as you wrote when you last touched this code) rather than `distributions.empty?`, that was a bug:

```
$ irb
irb(main):001:0> if []
irb(main):002:1>   puts "hi"
irb(main):003:1> end
hi
=> nil
```

Reply:
ok, makes sense, I did not realise `if []` is true in fact... strange it seemed to fix the issue I was having. thanks for info